### DISCUSSION TITLE
How to use a pkg-config port (zlib-ng) ?

### DISCUSSION THREAD

**Discussion by Guekka** [microsoft/vcpkg#19991] [Q&A] [answered]
Hey. I'm trying to use zlib-ng instead of zlib. But as far as I can see, it only provides a `.pc` file, not a CMake file.
`find_package(zlib-ng REQUIRED)` does not work, so how am I supposed to use it?
Thanks

**Comment by luncliff** [comment, marked as answer]
@Guekka, I'm the author of #18223. Try this.

```cmake
find_package(PkgConfig)
pkg_check_modules(zlib-ng REQUIRED IMPORTED_TARGET GLOBAL zlib-ng>=2.0)

target_link_libraries(main
PRIVATE
    PkgConfig::zlib-ng
)
```

This should result in CLI output like the following.

```
...
-- Checking for module 'zlib-ng>=2.0'
--   Found zlib-ng, version 2.0.3
...
```

**Reply by Guekka** [reply]
@luncliff Thank you, but it does not appear to work. 
It fails with the following message:
```
Could NOT find PkgConfig (missing: PKG_CONFIG_EXECUTABLE) 
CMake Error at C:/Program Files/CMake/share/cmake-3.21/Modules/FindPkgConfig.cmake:610 (message):
  pkg-config tool not found
Call Stack (most recent call first):
  C:/Program Files/CMake/share/cmake-3.21/Modules/FindPkgConfig.cmake:776 (_pkg_check_modules_internal)
  src/CMakeLists.txt:86 (pkg_check_modules)
```

I suppose vcpkg is somehow failing to make its internal PkgConfig available?

**Reply by dg0yt** [reply]
There is a gap between tools which are available when building ports and tools which are available in user projects.
You can build port `pkgconf` to get a suitable tool. But you need to help your user project to find the tool. Options, from specific to general:
- CMake variable `PKG_CONFIG_EXECUTABLE`
- environment variable `PKG_CONFIG`
- CMake variables that affect `find_program`
- environment variable `PATH`

**Comment by luncliff** [comment]
Nope. FindPkgConfig is a built in module in CMake. Try after install of the
tool. (maybe you have to configure PATH environment variable)
https://stackoverflow.com/q/1710922

**Comment by Guekka** [comment]
I see. This isn't very convenient, but I'll do with it. Thanks

**Reply by luncliff** [reply]
@Guekka, your message just reminded me that the port is using zlib compatible option. Would you try 'find_package(ZLIB)' instead of pkg-config?

**Reply by luncliff** [reply]
Just like what cmake supported for zlib 1.2.x

**Reply by Guekka** [reply]
`find_package(ZLIB REQUIRED)` results in an error, and I'm sure `zlib-ng` is installed. 
It looks like that feature was removed? https://github.com/microsoft/vcpkg/pull/18223#issuecomment-854833759

**Reply by luncliff** [reply]
You are right. It's removed. I confused with my working history. :exploding_head:
